src/core/public/chrome/chrome_service.tsx

```diff
@@ -94,11 +94,36 @@ export class ChromeService {
     injectedMetadata,
     notifications,
   }: StartDeps): Promise<InternalChromeStart> {
-    const FORCE_HIDDEN = isEmbedParamInHash();
+    /**
+     * These observables allow consumers to toggle the chrome visibility via either:
+     *   1. Using setIsVisible() to trigger the next chromeHidden$
+     *   2. Setting `chromeless` when registering an application, which will
+     *      reset the visibility whenever the next application is mounted
+     *   3. Having embed=true in the query string
+     */
+    const chromeHidden$ = new BehaviorSubject(false);
+    const forceHidden$ = of(isEmbedParamInHash());
```

Comment:
I don't know the actual usages of `setIsVisible` and `isEmbedParamInHash()` so the answer is probably 'yes', but I will ask anyway for my culture: is there actual cases where these two needs to be dissociated ? Will there be cases where the app starts with `isEmbedParamInHash()` being `true` and some code calling `setIsVisible(true)` afterwards ?

Reply:
Yeah it does seem like we could simplify this by just initializing `chromeHidden$` with `isEmbedParamInHash()`